core/types/block.go

```diff
@@ -477,6 +492,19 @@ func (b *Block) Hash() common.Hash {
 	return h
 }
 
+// SetVerkleProof attaches an execution witness + proof to the block in a verkle context.
+func (b *Block) SetVerkleProof(vp *verkle.VerkleProof, statediff verkle.StateDiff) {
```

Comment:
This should to be changed to `WithVerkleProof` and copy the block, like `WithBody`.